packages/loaders/src/Dots.spec.js

```diff
@@ -6,43 +6,228 @@
 
 describe('Dots', () => {
   beforeEach(() => {
     clearTimeout.mockClear();
+    global.cancelAnimationFrame = jest.fn();
+    global.requestAnimationFrame = jest.fn();
   });
 
-  describe('Rendering delay', () => {
-    it('renders LoadingPlacholder if delay has not been completed', () => {
-      const wrapper = mount(<Dots />);
+  describe('Loading delay', () => {
+    it('hides loader for initial delay', () => {
+      const { queryByTestId } = render(<Dots data-test-id="dots" />);
 
-      expect(wrapper.find(LoadingPlaceholder)).toExist();
+      expect(queryByTestId('dots')).toBeNull();
     });
 
-    it('does not render LoadingPlacholder if delay has completed', () => {
-      const wrapper = mount(<Dots />);
+    it('shows loader after initial delay', () => {
+      const { queryByTestId } = render(<Dots data-test-id="dots" />);
 
       jest.runOnlyPendingTimers();
-      wrapper.update();
-      expect(wrapper.find(LoadingPlaceholder)).not.toExist();
+
+      expect(queryByTestId('dots')).not.toBeNull();
     });
+  });
+
+  describe('Animation', () => {
+    it('updates animation after request animation frame', () => {
+      const { container } = render(<Dots data-test-id="dots" />);
+
+      jest.runOnlyPendingTimers();
 
-    it('clears delay timeout as component is unmounted', () => {
-      const wrapper = mount(<Dots />);
+      expect(container.firstChild.firstChild).toMatchInlineSnapshot(`
+<g
+  fill="currentColor"
+>
+  <circle
+    class=""
+    cx="9"
+    cy="9"
+    r="9"
+    transform="translate(31 32)"
+  />
+  <circle
+    class=""
+    cx="9"
+    cy="9"
+    r="9"
+    transform="translate(0 27.333333331333325)"
+  />
+  <circle
+    class=""
+    cx="9"
+    cy="9"
+    r="9"
+    transform="translate(62 27)"
+  />
+</g>
+`);
 
-      wrapper.unmount();
-      expect(clearTimeout).toHaveBeenCalledTimes(1);
+      // Requestion animation with 1000 MS delay
+      requestAnimationFrame.mock.calls[0][0](1000);
+
+      expect(container.firstChild.firstChild).toMatchInlineSnapshot(`
+<g
+  fill="currentColor"
+>
+  <circle
+    class="styled-elements__DotsCircle-sc-19dhio6-0 gEjoef"
+    cx="9"
+    cy="9"
+    r="9"
+    transform="translate(0 27)"
+  />
+  <circle
+    class="styled-elements__DotsCircle-sc-19dhio6-0 gEjoef"
+    cx="9"
+    cy="9"
+    r="9"
+    transform="translate(47.71816926427921 27)"
+  />
+  <circle
+    class="styled-elements__DotsCircle-sc-19dhio6-0 gEjoef"
+    cx="9"
+    cy="9"
+    r="9"
+    transform="translate(47.12454211563258 0.0012592386825828114)"
+  />
+</g>
+`);
     });
 
-    it('does not render LoadingPlaceholder if delayMS is 0', () => {
-      const wrapper = mount(<Dots delayMS={0} />);
+    it('updates animation after request animation frame with negative bound velocity', () => {
+      const { container } = render(<Dots velocity={-1.1} />);
+
+      jest.runOnlyPendingTimers();
+
+      expect(container.firstChild.firstChild).toMatchInlineSnapshot(`
```

Comment:
same for all the uses of `.firstChild.firstChild` I think querySelector is clearer as to what you're trying to find.